iotile_ext_cloud/iotile/cloud/cloud.py

```diff
@@ -88,6 +89,14 @@ def device_info(self, device_id):
 
         return dev
 
+    @return_type("bool")
+    def check_time(self):
+        """Checks if the current system time is consistent with iotile.cloud time
+        """
+        cloud_time = get_timestamp(requests.get('https://iotile.cloud').headers['Date'][5:-4])
```

Comment:
This should be more robust in case the headers change:

```
cloud_time = requests.get('').headers.get('Date', None)
if cloud_time is None:
    raise DataError("No date header returned from iotile.cloud")
```

Reply:
The custom string handling isn't necessary if we use dateutil.parser.parse as shown below and means we don't break if the string format changes for whatever reason in the future so long as dateutil supports it.

Reply:
I would not use that URL, as that may be a large one. Use instead `https://iotile.cloud/api/v1`.
In fact, do you guys rather we add a `https://iotile.cloud/api/v1/time/` @timburke @lelithium